tests/foreman/endtoend/test_api_endtoend.py

```diff
@@ -1228,21 +1227,21 @@ def test_positive_end_to_end(self, fake_manifest_is_set, default_sat, rhel6_cont
         # step 2.16: Create a new domain
         domain = entities.Domain(server_config).create()
 
-        # step 2.17: Create a new hostgroup and associate previous entities to
-        # it
+        # step 2.17: Create a new hostgroup and associate previous entities to it
         entities.HostGroup(server_config, domain=domain, subnet=subnet).create()
 
         # step 2.18: Provision a client
         # TODO this isn't provisioning through satellite as intended
         # Note it wasn't well before the change that added this todo
-        rhel6_contenthost.install_katello_ca(default_sat)
-        # Register client with foreman server using act keys
-        rhel6_contenthost.register_contenthost(org.label, activation_key_name)
-        assert rhel6_contenthost.subscribed
-        # Install rpm on client
-        package_name = 'python-kitchen'
-        result = rhel6_contenthost.execute(f'yum install -y {package_name}')
-        assert result.status == 0
-        # Verify that the package is installed by querying it
-        result = rhel6_contenthost.run(f'rpm -q {package_name}')
-        assert result.status == 0
+        with VMBroker(nick='rhel7', host_classes={'host': ContentHost}) as client:
```

Comment:
What's your thought on running this for multiple RHEL versions? If yes, use https://github.com/SatelliteQE/robottelo/blob/master/pytest_fixtures/core/contenthosts.py#L32

Reply:
@jyejare Yes, we can consider this in the future, maybe in 7.0 with single client tools repos.
currently, we've separate tools repos for each rhel, which will make this test more complicated.

Reply:
Please use `rhel7_contenthost` fixture for this purpose. Do not call broker directly from the test unless you have a strong reason for it.

Reply:
@Gauravtalreja1 No it should not be that challenging with ContentHost class supporting - https://github.com/SatelliteQE/robottelo/blob/master/robottelo/hosts.py#L228.

Reply:
@Gauravtalreja1 Since the test focuses on fixing the test, I wont force here to run it for all RHEL versions. @JacobCallahan @mshriver may confirm that we should soon have plans to run the client test on all RHEL versions across robottelo !